app/utils/deep_link/index.ts

```diff
@@ -114,37 +120,66 @@ export async function handleDeepLink(deepLinkUrl: string, intlShape?: IntlShape,
     }
 }
 
+const matcherOpts: Parameters<typeof match>[1] = {decode: decodeURIComponent};
+
+type ChannelPathParams = {
+    hostname: string;
+    subpath?: string;
+    teamName: string;
+    path: 'channels' | 'messages';
+    identifier: string;
+    postId?: string;
+};
+
+const CHANNEL_PATH = `/:subpath?/:teamName(${TEAM_NAME_PATH_PATTERN})/:path(channels|messages)/:identifier(${IDENTIFIER_PATH_PATTERN}|${ID_PATH_PATTERN})/:postId(${ID_PATH_PATTERN})?`;
```

Comment:
The subpath can be multiple path segments, right? This only matches one.

Reply:
Yes, refactored and found an additional bug in `matchDeepLink` related to this.